dynawo/sources/Solvers/SolverSIM/DYNSolverSIM.cpp

```diff
@@ -512,10 +512,12 @@ SolverSIM::solve() {
         skipNextNR_ = false;
         if (model_->getModeChangeType() != DIFFERENTIAL_MODE)
           return ROOT_ALG;
-        else if (getState().getFlags(ZChange) || getState().getFlags(SilentZChange))
+        else if (getState().getFlags(ZChange)
+            || getState().getFlags(SilentZNotUsedInDiscreteEqChange)
+            || getState().getFlags(SilentZNotUsedInContinuousEqChange))
```

Comment:
Here the behavior should be similar than below: if differential mode or other option if we change the logic then if ZChange or SilentZNotUsedInDiscreteEqChange ROOT else CONV.
This part should be refactored and simplified by #1255

Reply:
Solved